### ISSUE TITLE
Plan to migrate TCP Tracker from Data Tracker to ROW Portal

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1554]
Run of show:

- [x] disable TCP module in Data Tracker
- [x] migrate TCP tracking records
- [x] migrate TCP submission records
- [x] update connections between objects, accounts
- [x] migrate file attachments
- [x] switch ID auto increment fields from number to auto increments
- [ ] point TCP business days script to new app
- [x] activate accounts in ROW portal
- [x] send confirmation email to users

*Migrated from [atd-knack-data-tracker #231](https://github.com/cityofaustin/atd-knack-data-tracker/issues/231)*

**Comment by dianamartin** [comment]
I created the [ROW | Portal](https://atd.knack.com/row#home/)

**Comment by johnclary** [comment]
Had to abort the migration due to verrrrry slowwww indexing. I made a [copy of the app](https://builder.knack.com/atd/test-right-of-way--1-sep-2020) and am testing some optimizations for import.

**Comment by TracyLinder** [comment]
@johnclary @ChrispinP do the users need to be notified or updated?

**Comment by ChrispinP** [comment]
John and I have set the new Migration date for Wednesday September 9th. Will send out new email.

**Comment by johnclary** [comment]
Will try do another test migration with a subset of TCP data from 2021. Depending on how smoothly that goes, we'll decide if we want to refactor the way we manage statuses in the app or push ahead with the migration.

**Comment by johnclary** [comment]
@TracyLinder @ChrispinP I tested out my migration code. It's OK, but the record import functionality in the new Knack builder makes this a lot easier. Good call @ChrispinP! I created yet another version of this app [here](https://builder.knack.com/atd/right-of-way-portal-johns-migration-copy).

This app replicates the functionality of the existing TCP module in the Data Tracker, with a few exceptions.

- It does contain any of the automated logic which sets the status of parent TCP record based on the submission statuses. 
- It does not contain any of the date calculation logic which was used primarily for reporting.

I excluded this functionality because it's going to make the migration a lot easier. This is how I suggest we move forward:
- Have some TCP power users test out the new app.
- In parallel, get SSO login working for this app

Once we have SSO setup and users are OK with the functionality, we'll need to take the app offline on a Friday afternoon through the following Monday. The migration will entail:
- Importing all data
- Renaming all fields to be human friendly. Doing this after we migrate will make the imports easier.
- create and configure fields to generate record IDs (these slow down the migration, which is why we'll wait until data is imported)
- Run a script that uploads all the submission attachments.

Some things we need to discuss with ROW staff:
- Reporting: what reports are you using today, what reports do you need? We should strive to offload this to PowerBI
- We have a script that runs and automatically calculates 9 business days from the date of a submission. I believe it's used for reporting purposes. I'd like to make sure this is still needed, because we will need to re-configure this script to work with the new app

**Comment by TracyLinder** [comment]
**Next Steps**
1. Get answers to John's questions above
2. Finish the prototype
3. Get stakeholder feedback on prototype
4. Tracy/John/Chrispin sync

**Comment by johnclary** [comment]
@TracyLinder can you schedule some time with one or two TCP analysts so that we can test the new prototype?

Once we get their OK, we'll schedule the migration. Chrispin will migrate the records, and I'll follow-up by migrating the attachment files.

**Comment by ChrispinP** [comment]
SSO is complete #5993 and Pages/Forms have been updated to be user friendly to test existing functionality before migration.

Ready for Testing

**Comment by TracyLinder** [comment]
Per John: We should do user testing of the prototype with a couple of power users to make sure all features work (smoke test). 

- [ ] Tracy needs to schedule

**Comment by TracyLinder** [comment]
@ChrispinP can you please comment with the current status of this effort? Thank you.

**Comment by ChrispinP** [comment]
Migration confirmation sent. New ROW Portal is live. Will create a new issue for any followup/cleanup. Closing.
